src/kibana/components/vislib/visualizations/tile_map.js

```diff
@@ -321,6 +483,63 @@ define(function (require) {
     };
 
     /**
+     * Type of data overlay for map:
+     * creates canvas layer from mapData (geoJson)
+     * with leaflet.heat plugin
+     *
+     * @method heatMap
+     * @param map {Object}
+     * @param mapData {mapData Object}
+     * @return featureLayer {Leaflet object}
+     */
+    TileMap.prototype.heatMap = function (map, mapData) {
+      var self = this;
+      var max = mapData.properties.allmax;
+      var points = this.dataToHeatArray(mapData, max);
+
+      // default heatmap options in kibana:
+      // radius: 25,
+      // blur: 15,
+      // max: 1,
+      // minOpacity: 0.1,
+      // maxZoom: 18,
+      // gradient: {0.4: 'blue', 0.6: 'cyan', 0.7: 'lime', 0.8: 'yellow', 1: 'red'}
+
+      var options = {
+        radius: +this._attr.heatRadius,
+        blur: +this._attr.heatBlur,
+        maxZoom: +this._attr.heatMaxZoom,
+        minOpacity: +this._attr.heatMinOpacity
+      };
+
+      var featureLayer = L.heatLayer(points, options);
+
+      // add legend
+      //if (mapData.features.length > 1) {
+      //self.addHeatLegend(mapData, map);
```

Comment:
We try to avoid commenting out code. We can always go digging in git if we need it back: https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#do-not-comment-out-code